GeomIO/pFIOServerBounds.F90

```diff
@@ -0,0 +1,120 @@
+#include "MAPL_Generic.h"
+module mapl3g_pFIOServerBounds
+   use mapl_ErrorHandlingMod
+   use esmf
+   use pfio
+   use gFTL2_StringVector
+   use MAPL_BaseMod
+
+   implicit none
+   private
+
+   public :: pFIOServerBounds
+
+   integer, parameter :: grid_dims = 2
+
+   type :: pFIOServerBounds
+      private
+      integer, allocatable :: local_start(:)
+      integer, allocatable :: global_start(:)
+      integer, allocatable :: global_count(:)
+      integer, allocatable :: file_shape(:)
+   contains
+      procedure :: create_server_bounds
+      procedure :: get_local_start
+      procedure :: get_global_start
+      procedure :: get_global_count
+      procedure :: get_file_shape
+   end type pFIOServerBounds
+
+   contains
+
+   function get_local_start(this) result(local_start)
+      integer, allocatable :: local_start(:)
+      class(pFIOServerBounds), intent(in) :: this
+      local_start =this%local_start
+   end function get_local_start
+
+   function get_global_start(this) result(global_start)
+      integer, allocatable :: global_start(:)
+      class(pFIOServerBounds), intent(in) :: this
+      global_start =this%global_start
+   end function get_global_start
+
+   function get_global_count(this) result(global_count)
+      integer, allocatable :: global_count(:)
+      class(pFIOServerBounds), intent(in) :: this
+      global_count =this%global_count
+   end function get_global_count
+
+   function get_file_shape(this) result(file_shape)
+      integer, allocatable :: file_shape(:)
+      class(pFIOServerBounds), intent(in) :: this
+      file_shape =this%file_shape
+   end function get_file_shape
+
+   subroutine create_server_bounds(this, grid, field_shape, time_index, rc)
+      class(pFIOServerBounds), intent(inout) :: this
+      type(ESMF_Grid), intent(in) :: grid
+      integer, intent(in) :: field_shape(:)
+      integer, intent(in), optional :: time_index
+      integer, intent(out), optional :: rc
+
+      integer :: status, tile_count, n_dims, ungrid_dims, tm, global_dim(3)
+      integer :: i1, in, j1, jn, tile
+
+      call ESMF_GridGet(grid, tileCount=tile_count, _RC)
+      call MAPL_GridGetInterior(grid, i1,in, j1, jn)
+      call MAPL_GridGet(grid, globalCellCountPerDim=global_dim, _RC)
+      n_dims = size(field_shape)
+      ungrid_dims = n_dims - grid_dims
+      tm = 0
+      if (present(time_index))  tm = 1
+
```

Comment:
I think you want another local variable:  "total_dims".   With that, the ALLOCATE's can be refactored out of the conditionals.   

But this is just the tip of potential refactorings to eliminate what is now a long conditional (with an else).   I'll set up a meeting to pair-refactor this.

Reply:
see latest commit